src/java/org/apache/cassandra/io/sstable/format/big/BigFormat.java

```diff
@@ -411,7 +415,8 @@ static class BigVersion extends Version
             hasCommitLogIntervals = version.compareTo("mc") >= 0;
             hasAccurateMinMax = version.matches("(m[d-z])|(n[a-z])"); // deprecated in 'nc' and to be removed in 'oa'
             hasLegacyMinMax = version.matches("(m[a-z])|(n[a-z])"); // deprecated in 'nc' and to be removed in 'oa'
-            hasOriginatingHostId = version.matches("(m[e-z])") || version.compareTo("nb") >= 0;
+            // When adding a new version you might need to add it here
+            hasOriginatingHostId = version.matches("(m[e-z])|(o[a-z])") || version.compareTo("nb") >= 0;
```

Comment:
It seems that the `o[a-z]` part in `version.matches("(m[e-z])|(o[a-z])")` is already covered by `version.compareTo("nb")`. So I think this can be simplified to:
```suggestion
            hasOriginatingHostId = version.matches("(m[e-z])") || version.compareTo("nb") >= 0;
```
We can also put the faster string comparison before the probably slower pattern matching:
```suggestion
            hasOriginatingHostId = version.compareTo("nb") >= 0 || version.matches("(m[e-z])");
```

Reply:
I'm really surprised about this one as I distinctly remember adding that fix after a hellish debugging session. That's why I added the comment `// When adding a new version you might need to add it here`.

I guess in some previous version of the code sthg made it not work but it works as expected now. Done